src/lib/exec-fetcher.js

```diff
@@ -0,0 +1,90 @@
+const fs = require('fs-extra')
```

Comment:
Most of the logic here was extracted from https://github.com/netlify/cli/blob/93468a3fc43db5d637b30b9cbb21042da8626962/src/utils/live-tunnel.js

And made generic so we can re-use it.